libs/pubsub/pubsub_test.go

```diff
@@ -27,16 +27,49 @@ func TestSubscribe(t *testing.T) {
 	defer s.Stop()
 
 	ctx := context.Background()
-	ch := make(chan interface{}, 1)
-	err := s.Subscribe(ctx, clientID, query.Empty{}, ch)
+	subscription, err := s.Subscribe(ctx, clientID, query.Empty{})
 	require.NoError(t, err)
 	err = s.Publish(ctx, "Ka-Zar")
 	require.NoError(t, err)
-	assertReceive(t, "Ka-Zar", ch)
+	assertReceive(t, "Ka-Zar", subscription.Out())
 
 	err = s.Publish(ctx, "Quicksilver")
 	require.NoError(t, err)
-	assertReceive(t, "Quicksilver", ch)
+	assertReceive(t, "Quicksilver", subscription.Out())
+}
+
+func TestSubscribeWithoutCapacity(t *testing.T) {
+	s := pubsub.NewServer()
+	s.SetLogger(log.TestingLogger())
+	s.Start()
+	defer s.Stop()
+
+	ctx := context.Background()
+	assert.Panics(t, func() {
+		s.Subscribe(ctx, clientID, query.Empty{}, -1)
```

Comment:
Is this checking two distinct error cases? Seems like the second Subscribe never runs here since the first panics? Can we separate these into their own TestXxx function to test for Subscribe errors? cases?